### PR TITLE
Solution for unit and integration tests kovalets

### PR DESCRIPTION
**Pull request by AndriyKovalets**
Added base solution for unit test and two example of unit tests.
Added base solution for integtation test and example of integtation tests

### CODE REVIEW

**Review by FeloniukM — APPROVED** [review]

File: Provis.UnitTests/Base/TestData.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using Provis.Core.Entities.UserEntity;
+
+namespace Provis.UnitTests.Base
+{
+    public class TestData
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Base/TestData.cs]: Let's change class name

File: Provis.UnitTests/Core/Services/UserServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        private UserService _userService;
+
+        private Mock<UserManager<User>> _userManagerMock;
+        private Mock<IRepository<User>> _userRepositoryMock;
+        private Mock<IRepository<InviteUser>> _inviteUserRepositoryMock;
+        private Mock<IEmailSenderService> _emailSenderServiceMock;
+        private Mock<IFileService> _fileServiceMock;
+        private Mock<IOptions<ImageSettings>> _imageSettingsMock;
+        private IMapper _mapper;
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/UserServiceTests.cs]: It should be mocked

File: Provis.UnitTests/Core/Services/UserServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            _userRepositoryMock
+                .Setup(x => x.GetByKeyAsync(userId))
+                .ReturnsAsync(userMock);
+
+            var expectedUser = _mapper.Map<UserPersonalInfoDTO>(userMock);
+
+            var result = await _userService.GetUserPersonalInfoAsync(userId);
+
+            Assert.NotNull(result);
+            Assert.AreEqual(expectedUser.Email, result.Email);
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/UserServiceTests.cs]: Let's use fluentAssertions

File: Provis.UnitTests/Core/Services/UserServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+        [Test]
+        public async Task GetUserPersonalInfoAsync_UserExist_ReturnUserPersonalInfoDTO()
+        {
+            var userId = "1";
+            var userMock = TestData.GetTestUser();
+
+            _userRepositoryMock
+                .Setup(x => x.GetByKeyAsync(userId))
+                .ReturnsAsync(userMock);
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/UserServiceTests.cs]: Please add verifiable here

File: Provis.UnitTests/Core/Services/UserServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            Assert.NotNull(result);
+            Assert.AreEqual(expectedUser.Email, result.Email);
+        }
+
+        [Test]
+        public Task GetUserPersonalInfoAsync_UserNotExist_ThrowHttpException()
+        {
+            var userMock = TestData.GetTestUser();
+
+            _userRepositoryMock
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/UserServiceTests.cs]: Please add verifiable here

File: Provis.UnitTests/Core/Services/UserServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                .Setup(x => x.GetByKeyAsync(It.IsAny<string>()))
+                .Returns(Task.FromResult<User>(null));
+
+            var exeption = Assert.ThrowsAsync<HttpException>(() => _userService.GetUserPersonalInfoAsync("1"));
+
+            Assert.NotNull(exeption);
+            Assert.AreEqual(exeption.StatusCode, HttpStatusCode.NotFound);
+            Assert.AreEqual(exeption.Message, "User with Id not exist");
+
+            return Task.CompletedTask;
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/UserServiceTests.cs]: Let's use fluentAssertions

File: Provis.UnitTests/Core/Services/UserServiceTests.cs [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        }
+
+        [Test]
+        public Task GetUserPersonalInfoAsync_UserNotExist_ThrowHttpException()
+        {
+            var userMock = TestData.GetTestUser();
+
+            _userRepositoryMock
+                .Setup(x => x.GetByKeyAsync(It.IsAny<string>()))
+                .Returns(Task.FromResult<User>(null));
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/UserServiceTests.cs]: Let's move it into SetupMethod

File: Provis.UnitTests/Core/Services/UserServiceTests.cs [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+using Provis.Core.Interfaces.Repositories;
+using Provis.Core.Interfaces.Services;
+using Provis.Core.Services;
+using Provis.UnitTests.Base;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Provis.UnitTests.Core.Services
+{
+    [TestFixture]
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/UserServiceTests.cs:23]: Let's use terar down and verify all mocks

File: Provis.UnitTests/Base/MockUserManager.cs [hunk 9]
```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Moq;
+using System.Collections.Generic;
+
+namespace Provis.UnitTests.Base
+{
+    public class MockUserManager
+    {
+        public static Mock<UserManager<TUser>> GetUserManager<TUser>()
+        where TUser : class
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Base/MockUserManager.cs]: tabulation here

File: Provis.UnitTests/Base/MapperForTests.cs [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+namespace Provis.UnitTests.Base
+{
+    public class MapperForTests
+    {
+        public static IMapper GetMapper()
+        {
+            var mapperConfig = new MapperConfiguration(x =>
+               x.AddProfile(new ApplicationProfile()));
+
+            return mapperConfig.CreateMapper();
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Base/MapperForTests.cs]: We should add test for mapper and use mock of mapper for all other test
Reason is following the reason of failing test should be just one

File: Provis.UnitTests/Base/MockUserManager.cs (hunk 9 above)
Comment (AndriyKovalets) [reply Provis.UnitTests/Base/MockUserManager.cs]: Tabulation is added

File: Provis.UnitTests/Core/Services/UserServiceTests.cs (hunk 8 above)
Comment (AndriyKovalets) [reply Provis.UnitTests/Core/Services/UserServiceTests.cs:23]: terar down method is added

**Review by YevheniiVovk — APPROVED** [review]

**Review by YevheniiVovk — APPROVED** [review]

**Pull request merged**